pkg/controllers/options/tagging_controller.go

```diff
@@ -37,6 +37,13 @@ func (o *TaggingControllerOptions) Validate() error {
 		return fmt.Errorf("--tags must not be empty and must be a form of key=value")
 	}
 
+	for key := range o.Tags {
+		// We label the nodes with the tag keys for a cache-hit check so restrict tag keys to 63 characters.
+		if len(key) > 63 {
+			return fmt.Errorf("Tag keys must not be more than 63 characters long")
```

Comment:
Remind me -- does this cause the controller to exit?  And does it cause the entire CCM to crash if one controller exits?

Reply:
Yes, it would and on thinking again, it is probably not desirable since some customers might have a dynamic tag which gets set during runtime and we don't want it to cause an unrecoverable failure. Like I mentioned earlier, changing to a controller-specific label should fix this problem.